pkg/volume/flexvolume/api/objects.go

```diff
@@ -0,0 +1,72 @@
+package api
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// Probe plugin
+
+type FlexVolumeDriverCapabilities struct {
+	AccessMode          api.PersistentVolumeAccessMode
+	DynamicProvisioning bool
+	Attachment          bool
```

Comment:
Can we make Attachment an enum, instead of having two bools (Attachment and LocalAttachment, with LocalAttachment only being valid if Attachment is true).
I suggest Attachment is an enum, with 3 values :
1. None (same as Attachment = false)
2. Local (same as Attachment = true, LocalAttachment = true)
3. Remote (same as Attachment = true, LocalAttachment = false)

One less field to worry about, and one less case to validate :)